include/aligator/core/traj-opt-problem.hpp

```diff
@@ -101,6 +101,8 @@ template <typename _Scalar> struct TrajOptProblemTpl {
   ConstraintStackTpl<Scalar> term_cstrs_;
   /// Dummy, "neutral" control value.
   VectorXs unone_;
+  /// Function to initialize the state trajectory. If not set, uses xs_default_init.
+  std::function<void(const TrajOptProblemTpl &, std::vector<VectorXs> &)> xs_init_;
```

Comment:
You should maybe introduce a typedef for this function type (if we decide to keep using the type-erased `std::function`)